packages/builder/src/stores/builder/components.ts

```diff
@@ -1085,6 +1085,40 @@ export class ComponentStore extends BudiStore<ComponentState> {
     await this.patch(this.updateComponentSetting(name, value))
   }
 
+  isDatasourceUpdated(
+    component: Component,
+    setting: ComponentSetting | undefined,
+    update: any
+  ) {
+    if (setting?.type !== "dataSource") {
+      return false
+    }
+
+    const identifierKey: Record<string, string> = {
+      provider: "providerId",
+      query: "_id",
+      table: "resourceId",
+      viewV2: "resourceId",
+      view: "tableId", // Legacy
+    }
+    const currentSetting = component[setting!.key]
+    const resourceId = identifierKey[update.type]
+
+    // Legacy support.
+    let viewNameChange = false
+    if (currentSetting?.type === "view" && update.type === "view") {
+      // Could have the same tableId but the view name is different
+      viewNameChange = currentSetting.name !== update.name
+    }
+
+    return (
+      !currentSetting ||
+      currentSetting.type !== update.type ||
+      currentSetting?.[resourceId] !== update[resourceId] ||
+      viewNameChange
+    )
+  }
```

Comment:
Could we simplify this right down since it's only queries that have these extra params? I'm basically thinking that if the newly selected datasource type is anything other than `query` we can immediately return true that it changed. If the new type is `query`, compare the type and `_id` property with the previous value.

Reply:
Yep, agreed 👍  My first approach was exactly that; only catering to the `query._id` case as no other datasources allowed additional config. The only reason I went with this approach was future proofing. I'm happy to remove the other cases.